sdk/iptables/iptables.go

```diff
@@ -0,0 +1,158 @@
+package iptables
+
+import (
+	"errors"
+	"fmt"
+	"os/exec"
+	"strconv"
+)
+
+const (
+	// Chain to intercept inbound traffic
+	ProxyInboundChain = "PROXY_INBOUND"
+
+	// Chain to redirect inbound traffic to the proxy
+	ProxyInboundRedirectChain = "PROXY_IN_REDIRECT"
+
+	// Chain to intercept outbound traffic
+	ProxyOutputChain = "PROXY_OUTPUT"
+
+	// Chain to redirect outbound traffic to the proxy
+	ProxyOutputRedirectChain = "PROXY_REDIRECT"
+)
+
+// Config is used to configure which traffic interception and redirection
+// rules should be applied with the iptables commands.
+type Config struct {
+	// ProxyUserID is the user ID of the proxy process.
+	ProxyUserID string
+
+	// ProxyInboundPort is the port of the proxy's inbound listener.
+	ProxyInboundPort int
+
+	// ProxyInboundPort is the port of the proxy's outbound listener.
+	ProxyOutboundPort int
+
+	// IptablesProvider is the Provider that will apply iptables rules.
+	IptablesProvider Provider
+}
+
+// Provider is an interface for executing iptables rules.
+type Provider interface {
+	// AddRule adds a rule without executing it.
+	AddRule(name string, args ...string)
+	// ApplyRules executes rules that have been added via AddRule.
+	ApplyRules() error
```

Comment:
It could be good to document that this operation isn't atomic, and you could end up with some rules applied but not others if there's an error.

Reply:
Yep, will do! Although, I'll change it later to be atomic to better support VMs.